### ISSUE TITLE
Azure - Bad Request when attempting to tag App Service Plan

### ISSUE THREAD

**Issue by joescars** [cloud-custodian/cloud-custodian#3327]
**Issue** 

When trying to run policy to tag app service plans, custodian fails with the following error:

```
Error: 400 Client Error: Bad Request for url: https://management.azure.com/subscriptions/[SUBSCRIPTION_ID]/resourceGroups/[RESOURCE_GROUP]/providers/Microsoft.Web/serverfarms/[RESOURCE_NAME]?api-version=2018-11-01
2019-01-09 13:18:48,598: custodian.azure.utils.TagHelper:ERROR Failed to update tags for the resource.
Type: Microsoft.Web/serverfarms.
Name: EastUSPlan.
```

**Steps to Reproduce**

Attempt to run the following policy against your subscription (note you must have active app service plans)

```yaml
policies:
    - name: tag-app-service-plan
      description: |
        Adds a tag to all app service plans
      resource: azure.appserviceplan
      actions:
       - type: tag
         tag: Hello
         value: World
```
**Root Cause**

The Azure API requires a 'sku' property in the payload which is not provided in the API call. This can be empty but it is still required

**Proposed Resolution**

it **tools/c7n_azure/tags.py** change this 

https://github.com/cloud-custodian/cloud-custodian/blob/29225dd4740c98c7480f71025f49c67e11ec252b/tools/c7n_azure/c7n_azure/tags.py#L48-L53

to this:

```python
            generic_resource = GenericResource(location=az_resource.location,
                                               tags=tags,
                                               properties=az_resource.properties,
                                               kind=az_resource.kind,
                                               managed_by=az_resource.managed_by,
                                               identity=az_resource.identity,
                                               sku={})
```

I've tested locally and it works. It does not seem to affect any other type of call. 

Thoughts?

@aluong @logachev @luisalvesmartins @stefangordon

**Comment by erwelch** [comment]
@joescars what happens to the existing sku when updating with this empty object?

**Comment by joescars** [comment]
@erwelch it does not change. Here is an example 

request body

```json
{
  "id": "/subscriptions/[subscriptionid]/resourceGroups/ServerlessOpenHack/providers/Microsoft.Web/serverfarms/serverlessopenhack-jmr",
  "name": "serverlessopenhack-jmr",
  "type": "Microsoft.Web/serverfarms",
  "kind": "app",
  "sku": {},
  "properties": {
    "tags": {
    	"joe" : "raio",
    	"hello" : "world"
    }
  }
}
```
response body

```json
{
    "id": "/subscriptions/[subscriptionid]/resourceGroups/ServerlessOpenHack/providers/Microsoft.Web/serverfarms/serverlessopenhack-jmr",
    "name": "serverlessopenhack-jmr",
    "type": "Microsoft.Web/serverfarms",
    "kind": "functionapp",
    "tags": {
        "joe": "raio",
        "hello": "world"
    },
    "properties": {
        "serverFarmId": 20990,
        "name": "serverlessopenhack-jmr",
        "sku": {
            "name": "Y1",
            "tier": "Dynamic",
            "size": "Y1",
            "capacity": 0
        },
        "workerSize": "Default",
        "workerSizeId": 0,
        "workerTierName": null,
        "numberOfWorkers": 0,
        "currentWorkerSize": "Default",
        "currentWorkerSizeId": 0,
        "currentNumberOfWorkers": 0,
        "status": "Ready",
        "webSpace": "ServerlessOpenHack-SouthCentralUSwebspace",
        "subscription": "[subscriptionid]",
        "adminSiteName": null,
        "hostingEnvironment": null,
        "hostingEnvironmentProfile": null,
        "maximumNumberOfWorkers": 0,
        "planName": "VirtualDedicatedPlan",
        "adminRuntimeSiteName": null,
        "computeMode": "Dynamic",
        "siteMode": null,
        "geoRegion": "South Central US",
        "perSiteScaling": false,
        "maximumElasticWorkerCount": 0,
        "numberOfSites": 5,
        "hostingEnvironmentId": null,
        "isSpot": false,
        "spotExpirationTime": null,
        "freeOfferExpirationTime": null,
        "tags": {
            "joe": "raio",
            "hello": "world"
        },
        "kind": "functionapp",
        "resourceGroup": "ServerlessOpenHack",
        "reserved": false,
        "isXenon": false,
        "hyperV": false,
        "mdmId": "waws-prod-sn1-045_20990",
        "targetWorkerCount": 0,
        "targetWorkerSizeId": 0,
        "provisioningState": null,
        "webSiteId": null
    },
    "sku": {
        "name": "Y1",
        "tier": "Dynamic",
        "size": "Y1",
        "capacity": 0
    }
}
```

**Comment by logachev** [comment]
I think we actually should file an issue with Azure. rather than with custodian.
sku should be an optional parameter.
We can start with Azure Python SDK (https://github.com/Azure/azure-sdk-for-python). They can redirect it to the right person.

**Comment by stefangordon** [comment]
I remember struggling with SKU when I was testing out the `PATCH` call (which is what we are using now with the update_by_id).    The docs do include it in the request, but I thought the idea behind PATCH was ignoring anything which wasn't passed.

https://docs.microsoft.com/en-us/rest/api/resources/resources/updatebyid

At a minimum the fact that the behavior is different for this resource vs others indicates some lack of consistency we periodically see between different teams API implementations, so it is worth raising the issue at the SDK repo to ask if this is expected and if its safe to pass an empty one.

**Comment by stefangordon** [comment]
I believe this was fixed with #4494
